gcp/iam/workload_identity_federation/locals.tf

```diff
@@ -0,0 +1,43 @@
+locals  {
+  trusted_issuers_file = "${path.module}/trusted_issuers.yaml"
+
+  identity_pool_providers_map_trusted = {
+    for provider, specs in var.workload_identity_pool.providers: provider =>
+      contains(keys(yamldecode(file(local.trusted_issuers_file))), try(specs.oidc.issuer, "null")) ? yamldecode(
+        templatefile(local.trusted_issuers_file, {
+          owner = specs.owner
+          workspace_uuid = specs.workspace_uuid
+        } ))[specs.oidc.issuer] : null
+  }
```

Comment:
For each of the providers specified in `var.workload_identity_pool.providers`, this variable determines which of the preconfigured issuers is used (if any) from the `trusted_issuer_file`, and substitutes the `owner` and `workspace_uuid` values specified from the provider.
E.g. if a provider configured `oidc.issuer` as `bitbucket-pipelines`, this variable checks to see if `bitbucket-pipelines` is present in `local.trusted_issuers_file`, as it is it substitutes `owner` and `workspace_uuid` values using the `templatefile` function, and then maps the provider name to the `bitbucket-pipelines` configuration, giving a map like:
```terraform
identity_pool_provider_map_trusted = {
  bitbucket-provider = {
    issuer = "https://api.bitbucket.org/2.0/workspaces/SUBSITUTED_OWNER_VALUE/pipelines-config/identity/oidc"
    allowed_audiences = ["ari:cloud:bitbucket::workspace/SUBSTITUED_WORKPACE_UUID_VALUE"]
      attributes = {
        "google.subject" = "assertion.sub"
        "attribute.workspace_uuid" = "assertion.workspaceUuid"
        "attribute.repository" = "assertion.repositoryUuid"
        "attribute.git_ref" = "assertion.branchName"
      }
      condition = "assertion.workspaceUuid=='SUBSTITUED_WORKPACE_UUID_VALUE'"
  }
}
```
If the `oidc.issuer` for a provider is not present in the `trusted_issuers_file` then the value for that provider will be null.

This is used for configuring provider values in the `identity_pool_providers` variable